### ISSUE TITLE
Weighting instances in ranking model using xgboost

### ISSUE THREAD

**Issue by dengjn** [dmlc/xgboost#2561]
Does anyone know if I can weight instances in ranking model ('objective': 'rank:pairwise') in XGBoost? I know R GBM can only weight the whole query group instead of each instance within the query group. Does XGBoost also have this limitation? 

thanks,
jd

**Comment by pommedeterresautee** [comment]
related to https://github.com/dmlc/xgboost/issues/2231

**Comment by dengjn** [comment]
Thank you! But I mean how to use weight to calculate the loss function in ranking model, not use weight to calculate the evaluation metric such as mrr, ndcg. Does anyone have experience in this?

**Comment by ngoyal2707** [comment]
@dengjn https://github.com/dmlc/xgboost/pull/3379 PR should solve this issue, which is now merged to master.
